aws/subscriber_test.go

```diff
@@ -185,3 +185,34 @@ func TestEnsureFilterPolicy_DifferentPolicy_SetsSubscriptionAttributes(t *testin
 		t.Error("expected SetSubscriptionAttributes to be called, but wasn't")
 	}
 }
+
+// TestDeleteSubscription verifies that a sqs queue can be deleted successfully
+func TestDeleteSubscription(t *testing.T) {
+	sqsMock := mockSQS{}
+	s, serr := newSubscriber(&mockSNS{}, &sqsMock, "topic", "subscriptionID")
+	if serr != nil {
+		t.Errorf("expected no error from newSubscriber, but got: %v", serr)
+	}
+
+	sqsMock.stubbedDeleteQueueError = errors.New("test error")
+	expected := s.DeleteSubscription()
+	s.queueURL = aws.String("pubsub__topic-subscriptionID") // Set the Queue URL
+	sqsMock.spiedDeleteQueueInput = &sqs.DeleteQueueInput{QueueUrl: aws.String("pubsub__topic-subscriptionID")}
```

Comment:
hmm... why are you setting the spied input? This is what you want to verify is called.